File: NesneProg/Person/Person.py
class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age

    def get_name(self):
        return self.name

    def set_name(self, name):
        self.name = name

    def get_age(self):
        return self.age

    def set_age(self, age):
        self.age = age

    def display(self):
        print(f"Name: {self.name}\nAge: {self.age}")


class Student(Person):
    def __init__(self, name, age, student_id):
        super().__init__(name, age)
        self.student_id = student_id

    def get_student_id(self):
        return self.student_id

    def set_student_id(self, student_id):
        self.student_id = student_id

    def display(self):
        super().display()
        print(f"Student ID: {self.student_id}")


class Teacher(Person):
    def __init__(self, name, age, subject):
        super().__init__(name, age)
        self.subject = subject

    def get_subject(self):
        return self.subject

    def set_subject(self, subject):
        self.subject = subject

    def display(self):
        super().display()
        print(f"Subject: {self.subject}")

if __name__ == "__main__":
    person1 = Person("Hakan Tunc", 46)
    print("Person:")
    person1.display()
    print()

    student1 = Student("Deniz Kaya", 20, "218480")
    print("Student:")
    student1.display()
    print()

    student2 = Student("Kaan Gunay", 21, "263551")
    print("Student:")
    student2.display()
    print()

    teacher1 = Teacher("Yeliz Can", 33, "History")
    print("Teacher:")
    teacher1.display()
    print()

File: NesneProg/Person/Person.cpp
#include <iostream>
#include <string>
using std::string;

class Person {
protected:
    string name;
    int age;

public:
    Person(string n, int a) : name(n), age(a) {}

    string getName() { return name; }
    void setName(string n) { name = n; }

    int getAge() { return age; }
    void setAge(int a) { age = a; }

    virtual void display() {
        std::cout << "Name: " << name << "\nAge: " << age << std::endl;
    }
};

class Student : public Person {
private:
    string studentID;

public:
    Student(string n, int a, string id) : Person(n, a), studentID(id) {}

    string getStudentID() { return studentID; }
    void setStudentID(string id) { studentID = id; }

    void display() override {
        Person::display();
        std::cout << "Student ID: " << studentID << std::endl;
    }
};

class Teacher : public Person {
private:
    string subject;

public:
    Teacher(string n, int a, string sub) : Person(n, a), subject(sub) {}

    string getSubject() { return subject; }
    void setSubject(string sub) { subject = sub; }

    void display() override {
        Person::display();
        std::cout << "Subject: " << subject << std::endl;
    }
};

int main() {
    Person person1("Hakan Tunc", 46);
    std::cout << "Person:" << std::endl;
    person1.display();
    std::cout << std::endl;

    Student student1("Deniz Kaya", 20, "218480");
    std::cout << "Ogrenci :" << std::endl;
    student1.display();
    std::cout << std::endl;

    Student student2("Kaan Gunay", 21, "263551");
    std::cout << "Ogrenci :" << std::endl;
    student2.display();
    std::cout << std::endl;

    Teacher teacher1("Yeliz Can", 33, "Tarih");
    std::cout << "Ogretmen:" << std::endl;
    teacher1.display();
    std::cout << std::endl;

    return 0;
}
